core/src/main/java/com/turn/ttorrent/common/ByteBufferPool.java

```diff
@@ -0,0 +1,51 @@
+package com.turn.ttorrent.common;
+
+import org.apache.commons.pool2.BasePooledObjectFactory;
+import org.apache.commons.pool2.PooledObject;
+import org.apache.commons.pool2.impl.DefaultPooledObject;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
+
+import java.nio.ByteBuffer;
+
+public class ByteBufferPool {
+
+    private final GenericObjectPool<ByteBuffer> pool;
+
+    public GenericObjectPool<ByteBuffer> getPool() {
+        return pool;
+    }
+
```

Comment:
Indeed, could just wrap/hide `borrowObject()` and `returnObject()`..., guess it's cleaner like this, I'll update the last commit